test_conformance/spirv_new/test_spirv_14.cpp

```diff
@@ -0,0 +1,213 @@
+//
+// Copyright (c) 2024 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "testBase.h"
+#include "spirvInfo.hpp"
+#include "types.hpp"
+
+#include <string>
+
+extern bool gVersionSkip;
+
+static int test_image_operand_helper(cl_device_id deviceID, cl_context context,
+                                     cl_command_queue queue, bool signExtend)
+{
+    const char* filename = signExtend ? "spv1.4/image_operand_signextend"
+                                      : "spv1.4/image_operand_zeroextend";
+    cl_image_format image_format = {
+        CL_RGBA,
+        signExtend ? CL_SIGNED_INT8 : CL_UNSIGNED_INT8,
+    };
+
+    cl_int error = CL_SUCCESS;
+
+    std::vector<cl_uchar> imgData({ 0x1, 0x80, 0xFF, 0x0 });
+    std::vector<cl_uint> expected;
+    for (auto v : imgData)
+    {
+        if (signExtend)
+        {
+            expected.push_back((cl_int)(cl_char)v);
+        }
+        else
+        {
+            expected.push_back(v);
+        }
+    }
+
+    clProgramWrapper prog;
+    error = get_program_with_il(prog, deviceID, context, filename);
+    SPIRV_CHECK_ERROR(error, "Failed to compile spv program");
+
+    clKernelWrapper kernel = clCreateKernel(prog, "read_image_test", &error);
+    SPIRV_CHECK_ERROR(error, "Failed to create spv kernel");
+
+    std::vector<cl_uint> h_dst({ 0, 0, 0, 0 });
+    clMemWrapper dst =
+        clCreateBuffer(context, CL_MEM_COPY_HOST_PTR,
+                       h_dst.size() * sizeof(cl_uint), h_dst.data(), &error);
+    SPIRV_CHECK_ERROR(error, "Failed to create dst buffer");
+
+    std::vector<cl_uchar> h_imgdata({ 0x1, 0x80, 0xFF, 0x0 });
```

Comment:
Can we reuse `imgData` to avoid duplicating the input data?

Reply:
Good idea. Done.